packages/sdk/src/messageExtension/executeWithSSO.ts

```diff
@@ -0,0 +1,128 @@
+ * @throws {@link ErrorCode|RuntimeNotSupported} when runtime is nodeJS.
+ *
+ * @returns A MessageExtension Response for the activity. If the logic not return any, return void instead.
+ */
+export async function executionWithToken(
+  context: TurnContext,
+  config: AuthenticationConfiguration,
+  scopes: string | string[],
+  logic?: (token: TeamsMsgExtTokenResponse) => Promise<any>
+): Promise<MessagingExtensionResponse | void> {
+  const valueObj = context.activity.value;
+  if (!valueObj.authentication || !valueObj.authentication.token) {
+    internalLogger.verbose("No AccessToken in request, return silentAuth for AccessToken");
+    return getSignInResponseForMessageExtension(new TeamsFx(IdentityType.User, config), scopes);
+  }
+  try {
+    const teamsfx = new TeamsFx(IdentityType.User, config).setSsoToken(
+      valueObj.authentication.token
+    );
+    const token: AccessToken | null = await teamsfx.getCredential().getToken(scopes);
+    const ssoTokenExpiration: number = parseJwt(valueObj.authentication.token).exp;
+    const tokenRes: TeamsMsgExtTokenResponse = {
+      ssoToken: valueObj.authentication.token,
+      ssoTokenExpiration: new Date(ssoTokenExpiration * 1000).toISOString(),
+      token: token!.token,
+      expiration: token!.expiresOnTimestamp.toString(),
+      connectionName: "",
+    };
+    if (logic) {
+      return await logic(tokenRes);
+    }
+  } catch (err) {
+    if (err instanceof ErrorWithCode && err.code === ErrorCode.UiRequiredError) {
+      internalLogger.verbose("User not consent yet, return 412 to user consent first.");
+      const response = { status: 412 };
+      await context.sendActivity({ value: response, type: ActivityTypes.InvokeResponse });
+      return;
+    }
+    throw err;
+  }
+}
+
+/**
+ * Users execute query in message extension with SSO or access token.
+ *
+ * @param {TurnContext} context - The context object for the current turn.
+ * @param {AuthenticationConfiguration} config - User custom the message extension authentication configuration.
+ * @param {string| string[]} scopes - The list of scopes for which the token will have access.
+ * @param {function} logic - Business logic when executing the query in message extension with SSO or access token.
+ *
+ * @throws {@link ErrorCode|InternalError} when User invoke not response to message extension query.
+ * @throws {@link ErrorCode|InternalError} when failed to get access token with unknown error.
+ * @throws {@link ErrorCode|TokenExpiredError} when SSO token has already expired.
+ * @throws {@link ErrorCode|ServiceError} when failed to get access token from simple auth server.
+ * @throws {@link ErrorCode|InvalidParameter} when scopes is not a valid string or string array.
+ * @throws {@link ErrorCode|RuntimeNotSupported} when runtime is nodeJS.
+ *
+ * @returns A MessageExtension Response for the activity. If the logic not return any, return void instead.
+ */
+export async function queryWithToken(
```

Comment:
Since npm package does not have namespace, I'm not sure whether we want to express the function is related to messaging extension via the function name. I don't have good idea on this, `handleMessagingExtensionQueryWithToken` may be too long...
cc @MuyangAmigo I'm not sure whether we want to split packages in the future (maybe 2.0?), I heard of some discussions on this.

Reply:
or like  queryInMessageExtensionWithToken

Reply:
Good point here, I feel like long name isn't a problem as long as it generates more clarity. I agree with your proposal just change messaging -> message. 

Aslo we do have some ideas to split packages into smaller ones, but it hasn't been decided yet in terms of how we split it. (Tab/Bot, Browser/Node)

Reply:
sure, thanks